flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/RowTimeUnboundedOver.scala

```diff
@@ -116,7 +116,7 @@ abstract class RowTimeUnboundedOver(
     // discard late record
     if (timestamp > curWatermark) {
       // ensure every key just registers one timer
-      ctx.timerService.registerEventTimeTimer(curWatermark + 1)
+      ctx.timerService.registerEventTimeTimer(timestamp)
```

Comment:
Registering multiple timers on the same timestamp (`curWatermark + 1`), means that only one timer exists that fires exactly once when a watermark is received.

By registering timers on different timestamps, we have many timers that all will fire when a watermark is received. I don't think this is what we want.

Reply:
@fhueske Thanks a lot for your comments. Your concern makes sense to me. I think the current implementation is ok under periodic watermark. But I'm not sure if it's optimal under punctuated watermark. We will perform some performance test for unbounded over under punctuated watermark and share the results.